src/v/kafka/protocol/flex_versions.cc

```diff
@@ -9,11 +9,113 @@
+#include "kafka/protocol/schemata/add_partitions_to_txn_request.h"
+#include "kafka/protocol/schemata/alter_client_quotas_request.h"
+#include "kafka/protocol/schemata/alter_configs_request.h"
+#include "kafka/protocol/schemata/alter_partition_reassignments_request.h"
+#include "kafka/protocol/schemata/api_versions_request.h"
+#include "kafka/protocol/schemata/create_acls_request.h"
+#include "kafka/protocol/schemata/create_partitions_request.h"
+#include "kafka/protocol/schemata/create_topics_request.h"
+#include "kafka/protocol/schemata/delete_acls_request.h"
+#include "kafka/protocol/schemata/delete_groups_request.h"
+#include "kafka/protocol/schemata/delete_records_request.h"
+#include "kafka/protocol/schemata/delete_topics_request.h"
+#include "kafka/protocol/schemata/describe_acls_request.h"
+#include "kafka/protocol/schemata/describe_client_quotas_request.h"
+#include "kafka/protocol/schemata/describe_configs_request.h"
+#include "kafka/protocol/schemata/describe_groups_request.h"
+#include "kafka/protocol/schemata/describe_log_dirs_request.h"
+#include "kafka/protocol/schemata/describe_producers_request.h"
+#include "kafka/protocol/schemata/describe_transactions_request.h"
+#include "kafka/protocol/schemata/end_txn_request.h"
+#include "kafka/protocol/schemata/fetch_request.h"
+#include "kafka/protocol/schemata/find_coordinator_request.h"
+#include "kafka/protocol/schemata/heartbeat_request.h"
+#include "kafka/protocol/schemata/incremental_alter_configs_request.h"
+#include "kafka/protocol/schemata/init_producer_id_request.h"
+#include "kafka/protocol/schemata/join_group_request.h"
+#include "kafka/protocol/schemata/leave_group_request.h"
+#include "kafka/protocol/schemata/list_groups_request.h"
+#include "kafka/protocol/schemata/list_offset_request.h"
+#include "kafka/protocol/schemata/list_partition_reassignments_request.h"
+#include "kafka/protocol/schemata/list_transactions_request.h"
+#include "kafka/protocol/schemata/metadata_request.h"
+#include "kafka/protocol/schemata/offset_commit_request.h"
+#include "kafka/protocol/schemata/offset_delete_request.h"
+#include "kafka/protocol/schemata/offset_fetch_request.h"
+#include "kafka/protocol/schemata/offset_for_leader_epoch_request.h"
+#include "kafka/protocol/schemata/produce_request.h"
+#include "kafka/protocol/schemata/sasl_authenticate_request.h"
+#include "kafka/protocol/schemata/sasl_handshake_request.h"
+#include "kafka/protocol/schemata/sync_group_request.h"
+#include "kafka/protocol/schemata/txn_offset_commit_request.h"
 #include "kafka/protocol/types.h"
-#include "kafka/server/handlers/handlers.h"
+#include "kafka/protocol/wire.h"
 #include "utils/vint_iostream.h"
+
+#include <seastar/core/iostream.hh>
+
 namespace kafka {
 
+template<typename... Ts>
+struct type_list {};
+
+template<typename... Requests>
+using make_request_types = type_list<Requests...>;
+
+using request_types = make_request_types<
+  produce_api,
```

Comment:
Notice that unlike the similar list in `src/v/kafka/server/handlers/handlers.h` these are the pure protocol metadata, unrelated to the server handlers.